core/region.go

```diff
@@ -330,22 +330,30 @@ func isASGWithMatchingTags(asg *autoscaling.Group, tagsToMatch []Tag) bool {
 	return matchedTags == len(tagsToMatch)
 }
 
-func (r *region) findMatchingASGsInPageOfResults(groups []*autoscaling.Group, tagsToMatch []Tag) []autoScalingGroup {
+func (r *region) findMatchingASGsInPageOfResults(groups []*autoscaling.Group,
+	tagsToMatch []Tag) []autoScalingGroup {
 
 	var asgs []autoScalingGroup
+	var optInFilterMode = (r.conf.TagFilteringMode == "" ||
+		r.conf.TagFilteringMode == "opt-in")
 
 	for _, group := range groups {
-		if isASGWithMatchingTags(group, tagsToMatch) {
-			asgName := *group.AutoScalingGroupName
-			logger.Println("Matching tags found for ASG, enabling ASG for processing:", asgName)
-			asgs = append(asgs, autoScalingGroup{
-				Group:  group,
-				name:   asgName,
-				region: r,
-			})
+		groupMatchesExpectedTags := isASGWithMatchingTags(group, tagsToMatch)
+		// Go lacks a logical XOR operator, this is the equivalent to that logical
+		// expression. The goal is to add the matching ASGs when running in opt-in
+		// mode and the other way round.
+		if optInFilterMode != groupMatchesExpectedTags {
+			continue
 		}
-	}
 
+		asgName := *group.AutoScalingGroupName
+		logger.Println("Matching tags found for ASG, enabling ASG for processing:", asgName)
```

Comment:
This message could now be a bit miss-leading, considering that in the `opt-out`, the lack of tag would be the reason for which they got selected.

Maybe using: `Tag (%s) and mode (%s) matching of this ASG` could do? Thus making explicit the tag/mode.